### PR TITLE
Cummerbund

### PR DESCRIPTION
**Pull request by bagnacan**
Modified training on "Visualization of RNA-Seq results with CummeRbund"
- Explained each field of the CuffDiff output that is used as input data
- Previous filter/sort procedure filtered on segnificance attribute, then sorted descending on log2fold_change. Since the goal is instead to sort on the _statistically assessed_ log2fold_change, the new filter/sort procedure filters on significance attribute, and then sorts on Q-value (ascending) and log2fold_change (descending)
- Added link to the ggplot objects generated by CummeRbund
- Added heatmap for the top-5 significant differentially expressed gene isoforms

### CODE REVIEW

File: topics/transcriptomics/tutorials/rna-seq-viz-with-cummerbund/tutorial.md [hunk 1]
```diff
@@ -116,98 +138,131 @@ For this tutorial, we are interested in CuffDiff's tested transcripts for differ
...
 >
 > 2. **Sort** {% icon tool %}: with the following parameters
->   - "Sort Dataset" to the filtered table
->   - "on column", "with flavor" and "everything in"  to the appropriate values (see above)
+>   - "Sort Dataset" to the previously filtered table
+>   - "on column", to 13
+>   - "with flavor" to "Numerical sort"
+>   - "everything in" to "Ascending order"
+>   - Click on "Insert Column selection", and parametrize the Sort tool to sort on column 10. Be careful of the sorting order!
+>   - Are there any rows allocated for the teable's header? In that case, set "Number of header lines to skip" accordingly!
```
Comment (shiltemann) [inline topics/transcriptomics/tutorials/rna-seq-viz-with-cummerbund/tutorial.md]:
```suggestion
>   - Are there any rows allocated for the table's header? In that case, set "Number of header lines to skip" accordingly!
```

File: topics/transcriptomics/tutorials/rna-seq-viz-with-cummerbund/tutorial.md [hunk 2]
```diff
@@ -116,98 +138,131 @@ For this tutorial, we are interested in CuffDiff's tested transcripts for differ
...
+>  We will sort all records on the basis of their Q-score (column 13) and log<sub>2</sub>(fold_change).
 >
 > 2. **Sort** {% icon tool %}: with the following parameters
->   - "Sort Dataset" to the filtered table
->   - "on column", "with flavor" and "everything in"  to the appropriate values (see above)
+>   - "Sort Dataset" to the previously filtered table
+>   - "on column", to 13
+>   - "with flavor" to "Numerical sort"
+>   - "everything in" to "Ascending order"
+>   - Click on "Insert Column selection", and parametrize the Sort tool to sort on column 10. Be careful of the sorting order!
```
Comment (shiltemann) [inline topics/transcriptomics/tutorials/rna-seq-viz-with-cummerbund/tutorial.md]:
```suggestion
>   - Click on "Insert Column selection", and parameterize the Sort tool to sort on column 10. Be careful of the sorting order!
```

Comment (nsoranzo) [reply topics/transcriptomics/tutorials/rna-seq-viz-with-cummerbund/tutorial.md]: Both forms are fine, I think.

**Comment by shiltemann** [comment]
thanks a lot @bagnacan! 

ping @mblue9 perhaps you would like to review this?

File: topics/transcriptomics/tutorials/rna-seq-viz-with-cummerbund/tutorial.md [hunk 3]
```diff
@@ -100,7 +100,29 @@ For this tutorial, we are interested in CuffDiff's tested transcripts for differ
...
+>   - "FAIL" if a numerical exception prevented testing
+> - *value_1*: The gene's FPKM in sample_1
+> - *value_2*: The gene's FPKM in sample_2
+> - *log2(fold_change)*: The log<sub>2</sub> of the fold change (sample_1/sample_2). This reports the expression difference between condition one (sample_1) and condition two (sample_2)
+> - *test_stat*: The test statistic's value, used to determine the significance of the observed change in FPKM
+> - *p_value*: The uncorrected p-value of the test statistic
+> - *q_value*: The False-discovery-rate-adjusted p-value of the test statistic
+> - *significant*: "yes" or "no", depending on whether the p_value is greater then the FDR after Benjamini-Hochberg correction for multiple-testing. This tells whether the difference between the expression levels in condition one (sample_1) and condition two (sample_2) is significant
+>
+> We want to keep only those the *significant differentially expressed* genes.
```
Comment (mblue9) [inline topics/transcriptomics/tutorials/rna-seq-viz-with-cummerbund/tutorial.md]: We want to keep only the *significant differentially expressed* genes.

File: topics/transcriptomics/tutorials/rna-seq-viz-with-cummerbund/tutorial.md [hunk 4]
```diff
@@ -116,53 +138,59 @@ For this tutorial, we are interested in CuffDiff's tested transcripts for differ
 
 # Filtering and sorting
 
-We now want to first highlight the most significant differentially expressed genes in our analysis, and then obtain informative visualizations.
+We now want to highlight those gene transcripts whose expression difference, the log<sub>2</sub>(fold_change), has been statistically assessed as both high and significant.
```
Comment (mblue9) [inline topics/transcriptomics/tutorials/rna-seq-viz-with-cummerbund/tutorial.md]: We now want to highlight the gene transcripts whose expression difference, the log<sub>2</sub>(fold_change), has been statistically assessed to be both high and significant.

File: topics/transcriptomics/tutorials/rna-seq-viz-with-cummerbund/tutorial.md [hunk 5]
```diff
@@ -116,53 +138,59 @@ For this tutorial, we are interested in CuffDiff's tested transcripts for differ
...
 >
 >    > ### {% icon question %} Questions
 >    > 1. What column stores the information of significance for each record?
 >    > 2. Which conditional expression has to be set to filter all records on the selected column?
->    > 3. What happened to the records in the original table?
+>    > 3. How many rows are allocated for the table's header?
+>    > 4. How many entries were originally stored in the table? And how many after the filtering operation?
 >    >
 >    > > ### {% icon solution %} Solution
 >    > > 1. column 14
```
Comment (mblue9) [inline topics/transcriptomics/tutorials/rna-seq-viz-with-cummerbund/tutorial.md]: s/column/Column

File: topics/transcriptomics/tutorials/rna-seq-viz-with-cummerbund/tutorial.md [hunk 6]
```diff
@@ -116,53 +138,59 @@ For this tutorial, we are interested in CuffDiff's tested transcripts for differ
...
+>    > > 3. 1
+>    > > 4. ~140.000 (before filtering) vs. 219 (after filtering)
 >    > {: .solution }
 >    {: .question}
 >
->  Look at your data. The differential expression values are stored on column 10, we will sort (descending) all records
->  on the basis of their value at the 10th column
+>  Review the meaning of each column, and look at your data.
+>
+>  - The differential expression values are stored on column 10
```
Comment (mblue9) [inline topics/transcriptomics/tutorials/rna-seq-viz-with-cummerbund/tutorial.md]: The differential expression values are stored in column 10

File: topics/transcriptomics/tutorials/rna-seq-viz-with-cummerbund/tutorial.md [hunk 7]
```diff
@@ -116,53 +138,59 @@ For this tutorial, we are interested in CuffDiff's tested transcripts for differ
...
+>  - The statistical score, assessing the differential expression significance, is stored on column 13
```
Comment (mblue9) [inline topics/transcriptomics/tutorials/rna-seq-viz-with-cummerbund/tutorial.md]: stored in column 13

File: topics/transcriptomics/tutorials/rna-seq-viz-with-cummerbund/tutorial.md (hunk 4 above)
Comment (mblue9) [reply topics/transcriptomics/tutorials/rna-seq-viz-with-cummerbund/tutorial.md]: or I think you can leave "statistically assessed as"

**Comment by bagnacan** [comment]
@shiltemann, @mblue9, thank you for your edits and suggestions! I addressed the last comments in my latest contribution. What do you think?

**Comment by shiltemann** [comment]
@bagnacan thanks for the edits and don't worry about dismissing the stale review, that happens automatically when changes are made after an approval :)

**Comment by mblue9** [comment]
Thanks @bagnacan, I just added some tiny edits.

**Pull request merged**